libnavigation-base/src/main/java/com/mapbox/navigation/base/route/NavigationV2Router.kt

```diff
@@ -0,0 +1,39 @@
+package com.mapbox.navigation.base.route
+
+import com.mapbox.api.directions.v5.models.DirectionsRoute
+import com.mapbox.navigation.base.CurrentIndicesSnapshot
+
+/**
+ * Extends [NavigationRouter] to also provide ability for refreshing routes partially
+ * using indices stored in [CurrentIndicesSnapshot].
+ */
+interface NavigationV2Router : NavigationRouter {
```

Comment:
Should we deprecate `NavigationRouter` then?

Reply:
We did not deprecate Router...

Reply:
oh, I thought we did 😮 
Let's wait for @LukasPaczos to tell us about migration from `Router` to `NavigationRouter` then  https://github.com/mapbox/mapbox-navigation-android/pull/6190#discussion_r949941522

Reply:
We didn't deprecate the `Router` because that's still the interface that developers can use for modularization - https://github.com/mapbox/mapbox-base-android/blob/d97bdcfe9ac43c2c6a63eb7b7a85456350ccd818/annotations/src/main/java/com/mapbox/annotation/module/MapboxModuleType.kt#L52-L55. As mentioned above, I imagine us moving away from a modularized router in the future, that's why we didn't invest in migration.